extensions/amp-story/1.0/amp-story.js

```diff
@@ -444,14 +444,19 @@ export class AmpStory extends AMP.BaseElement {
       return;
     }
     // The theme color should be copied from the story's primary accent color
-    // if possible, with the fall back being default light gray.
+    // if possible, with the fall back being default dark gray.
     const meta = this.win.document.createElement('meta');
     const ampStoryEl = this.win.document.getElementsByTagName('amp-story')[0];
-    const styles = computedStyle(this.win, ampStoryEl);
+    const ampStoryPageEl =
+      this.win.document.getElementsByTagName('amp-story-page')[0];
+    const storyStyles = computedStyle(this.win, ampStoryEl);
+    const pageStyles = computedStyle(this.win, ampStoryPageEl);
     meta.name = 'theme-color';
-    meta.content = styles.getPropertyValue('--primary-color') ||
-        DEFAULT_THEME_COLOR;
-    this.win.document.getElementsByTagName('head')[0].appendChild(meta);
+    meta.content =
+      storyStyles.getPropertyValue('--primary-color') ||
+      pageStyles.getPropertyValue('background-color') ||
+      DEFAULT_THEME_COLOR;
```

Comment:
Nit: can we avoid getting the computedStyle of the page, if we found what we were looking for in the `storyStyles`?
something like:
```
computedStyle(this.win, ampStoryEl).getPropertyValue('--primary-color') ||
computedStyle(this.win, ampStoryPageEl).get....
```